### DISCUSSION TITLE
Documentation rules

### DISCUSSION THREAD

**Discussion by frankystone** [widelands/widelands#4779] [Ideas]
I've started to make a wikiarticle about documentation rules: https://www.widelands.org/wiki/Documentation%20Rules/

Since i want to work on the documentation i have a question about string constants like e.g. for roadtypes: Currently they are formatted like ‘normal’, ‘busy’, or ‘waterway’. Since they are strings we should use another formatting imho. Something like `"normal"`, `"busy"`, or `"waterway"` ?

What do you think?

**Comment by Noordfrees** [comment]
+1 for surrounding them with `"quotes"`, otherwise some scripters might think these are global constants

**Comment by Noordfrees** [comment]
Since the API is intended to be backward-compatible for future versions it would be good to add a note in which version a feature was introduced for all functions added after v1.0.
Is there some standard way to add a `:since_version:` tag or something like this to a method/attribute description? Or does this need to go in the description text?

**Comment by frankystone** [comment]
According to the [doc](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded) you can use:

~~~~
.. versionadded:: version
   text

.. versionchanged:: version
   text

.. deprecated:: version
   text
~~~~

Not tested.

**Reply by Noordfrees** [reply]
Thanks :)

Looks good:
![grafik](https://user-images.githubusercontent.com/48293446/119861632-e316b000-bf17-11eb-936b-6f95c89ed3ae.png)

**Reply by frankystone** [reply]
whohoo... is there someone working on different types of ships? :+1: 

Don't forget real life. It's somewhere out of here :)
